include/op.h

```diff
@@ -92,12 +104,34 @@ typedef struct header_s {
   char comment[COMMENT_LENGTH + 1];
 } header_t;
 
-/* live */
+typedef struct champion
+{
+    header_t *champ_header;       // header
+    int id;                       // id of the champ
+    int num_instuctions;          // number of instructions
+    op_t *instructions;           // instruction array
```

Comment:
I need to rethink how the instructions will be stored for each champion.

Reply:
i agree but will go ahead and merge for now